File: src/src/gestorAplicacion/libreria/Titulo.java
package gestorAplicacion.libreria;

import java.io.Serializable;

public abstract class Titulo implements Serializable, Comparable<Titulo> {
    private String nombre;
    private String autor;
    private final int ISBN;
    private int usos = 0;

    public Titulo(String nombre, String autor, int ISBN) {
        this.nombre = nombre;
        this.autor = autor;
        this.ISBN = ISBN;
    }

    public Titulo(int isbn) {
        ISBN = isbn;
    }

    //getters y setters

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public int getISBN() {
        return ISBN;
    }


    public int getUsos() {
        return usos;
    }

    public void setUsos(int usos) {
        this.usos = usos;
    }

    //Metodo abstracto
    public abstract String mostrarse();

    public void usado(){
        usos++;
    }

    @Override
    public int compareTo(Titulo titulo) {
        if (titulo.getUsos()>usos){
            return 1;
        } else if (titulo.getUsos()>usos) {
            return 0;
        }else {
            return -1;
        }
    }
}

File: Python/gestorAplicacion/libreria/titulo.py


class Titulo():

    def __init__(self, nombre: str, autor: str, ISBN: int):
        self._nombre = nombre
        self._autor = autor
        self._ISBN = ISBN
        self._usos = 0

    def getNombre(self) -> str:
        return self._nombre

    def setNombre(self, nombre: str):
        self._nombre = nombre

    def getAutor(self) -> str:
        return self._autor

    def setAutor(self, autor: str):
        self._autor = autor

    def getISBN(self) -> int:
        return self._ISBN

    def getUsos(self) -> int:
        return self._usos

    def setUsos(self, usos: int):
        self._usos = usos

    def usado(self):
        self._usos += 1

    def mostrarse(self):
        pass